packages/dds/tree/src/core/tree/anchorSet.ts

```diff
@@ -447,7 +554,37 @@ class PathNode implements UpPath {
 		 * This consistency guarantee only applies to nodes that are `Alive`.
 		 */
 		public parentPath: PathNode | undefined,
-	) {}
+	) {
+		super(1);
+	}
+
+	public on<K extends keyof AnchorEvents>(eventName: K, listener: AnchorEvents[K]): () => void {
+		return this.events.on(eventName, listener);
+	}
+
+	child(key: FieldKey, index: number): UpPath<AnchorNode> {
+		// Fast path: if child exists, return it.
+		return (
+			this.tryGetChild(key, index) ?? { parent: this, parentField: key, parentIndex: index }
+		);
+	}
+
+	getOrCreateChildRef(key: FieldKey, index: number): [Anchor, AnchorNode] {
+		const anchor = this.anchorSet.track(this.child(key, index));
+		const node =
+			this.anchorSet.locate(anchor) ?? fail("cannot reference child that does not exist");
+		return [anchor, node];
+	}
+
+	slotMap<T extends AnchorKey<unknown>>(
```

Comment:
You'd have to expose deleteSlot (and maybe hasSlot, depending on signatures) as well, though.

Reply:
You would want has, delete, get and set. I mostly figured it would be simpler to just expose the map.

I dislike the API for maps as it forces redundant lookups (ex: getOrCreate has to do the look up twice), so I figured if I exposed a real map, then its not my fault the API is bad, and functions like our getOrCreate helpers can work on it, and we don't add overhead in the codepath for the re-lookup.

That said, I think this turned out to be a bad way to expose the map. I have updated the code with a better one.